contracts/src/systems/attack.cairo

```diff
@@ -1,7 +1,113 @@
 #[system]
 mod attack_system {
     use dojo::world::Context;
+    use traits::Into;
+    use traits::TryInto;
+    use option::{Option, OptionTrait};
+    use tsubasa::components::{Game, Card, Roles, Placement, Player};
+    use tsubasa::events::EndTurn;
+    use debug::PrintTrait;
 
-    fn execute(ctx: Context, player: u8) {}
+    /// * `ctx` - Dojo context.
+    /// * `game_id` - The current game_id.
+    /// * `card_id_player1` - The token id of the card placed.
+    /// * `card_id_player2` - The token id of the card placed.
+    fn execute(ctx: Context, game_id: felt252, card_id_player1: felt252, card_id_player2: felt252) {
+        let game = get!(ctx.world, game_id, Game);
+
+        let mut card_id_player1: u256 = card_id_player1.into();
+        let mut card_id_player2: u256 = card_id_player2.into();
+
+        let card_player1 = get!(ctx.world, card_id_player1, Card);
+        let card_player2 = get!(ctx.world, card_id_player2, Card);
+
+        let game_player_key: (felt252, felt252) = (game_id, ctx.origin.into());
+        let game_player2_key: (felt252, felt252) = (game_id, game.player2.into());
+
+        let mut player1 = get!(ctx.world, game_player_key, Player);
+        let mut player2 = get!(ctx.world, game_player2_key, Player);
+
+        // We set a value as boolean replacement to match the Roles
+        let mut value = 0;
+        match card_player1.role {
+            Roles::Goalkeeper => value = 0,
+            Roles::Defender => value = 0,
+            Roles::Midfielder => value = 0,
+            Roles::Attacker => value = 1,
+        }
+
+        if value == 1 {
```

Comment:
this is not right, you should loop through all the players card, they'll all attack one by one starting from the `Roles::Attacker`